### PR TITLE
new: Support NodeBalancers in Firewall-related resources

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This change adds support for attaching NodeBalancers to Cloud Firewalls through the `linode_firewall` and `linode_firewall_device` resources. Additionally, this change updates Firewall-related data sources to reflect the new changes.

Most of the files changed in this PR are small test adjustments.

## ✔️ How to Test

```
make PKG_NAME=linode/firewall testacc
make PKG_NAME=linode/firewalldevice testacc
```

### CODE REVIEW

File: linode/nb/framework_resource.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (r *Resource) Create(
 	}
 
 	if !data.Tags.IsNull() {
-		resp.Diagnostics.Append(data.Tags.ElementsAs(ctx, &createOpts.Tags, false)...)
+		resp.Diagnostics.Append(data.Tags.ElementsAs(ctx, &createOpts.Tags, true)...)
```
Comment (lgarber-akamai) [inline linode/nb/framework_resource.go:55]: This fixes an infinite diff issue that was causing tests in this PR to fail

File: go.mod [hunk 2]
```diff
@@ -95,4 +95,7 @@ require (
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
 
+// TODO: Remove this replacement before merging to dev
+replace github.com/linode/linodego => github.com/linode/linodego v1.21.1-0.20230905223015-1df9dbe40b16
+
```
Comment (zliang-akamai) [inline go.mod:100]: So this PR should target the feature branch?

Comment (lgarber-akamai) [reply go.mod:100]: So unfortunately `replace` don't support branch names with `/` in them so I had to pin to commit hash. Fortunately this feature has been merged and released in linodego, so I can drop this replace altogether.

Comment (zliang-akamai) [reply go.mod:100]: Oh, sorry about the confusion. I meant it said remove this before merging to dev but this PR is targeting dev, lol

Comment (lgarber-akamai) [reply go.mod:100]: Oh oops, good catch! I've re-targeted it to the feature branch.

**Review by yec-akamai — APPROVED** [review]
Nice work!

File: linode/firewalldevice/resource_test.go [hunk 3]
```diff
@@ -19,12 +20,12 @@ import (
...
-	if err != nil {
-		log.Fatal(err)
-	}
+	//region, err := acceptance.GetRandomRegionWithCaps([]string{"Cloud Firewall"})
+	//if err != nil {
+	//	log.Fatal(err)
+	//}
 
-	testRegion = region
+	testRegion = "us-east"
```
Comment (zliang-akamai) [inline linode/firewalldevice/resource_test.go:28]: Since we are targeting dev, let's restore these?

File: linode/firewall/resource_test.go [hunk 4]
```diff
@@ -8,45 +8,31 @@ import (
...
-	testProviders = providerMap
-
-	region, err := acceptance.GetRandomRegionWithCaps([]string{"Cloud Firewall"})
+	_, err := acceptance.GetRandomRegionWithCaps([]string{"Cloud Firewall"})
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	testRegion = region
+	testRegion = "us-east"
```
Comment (zliang-akamai) [inline linode/firewall/resource_test.go:35]: Also restore this?

File: linode/nb/framework_resource_schema.go [hunk 5]
```diff
@@ -77,6 +77,7 @@ var frameworkResourceSchema = schema.Schema{
 		"tags": schema.SetAttribute{
 			ElementType: types.StringType,
 			Optional:    true,
+			Computed:    true,
```
Comment (zliang-akamai) [inline linode/nb/framework_resource_schema.go:80]: I think `tags` isn't computed?

Comment (lgarber-akamai) [reply linode/nb/framework_resource_schema.go:82]: Nice catch! This was related to the infinite diff mentioned above, but I was able to come up with a better solution using defaults after taking another look at this issue.

**Review by zliang-akamai — APPROVED** [review]

**Pull request merged**